### ISSUE TITLE
Glitches when adding newlines

### ISSUE THREAD

**Issue by jcubic** [codemirror/codemirror5#6935]
I'm not sure if this is because of bugs in PEG.js mode but I have weird results when I'm adding newlines to the editor. Somethings all lines are broken sometimes they are fixed.

![PEGJS_mode](https://user-images.githubusercontent.com/280241/173076875-7d29cdf3-9561-4a70-8dcf-669162d95ebe.gif)

This may be because incremental parser and PEG.js have inner JS mode.

Do you think that the problem is in PEG.js mode or in CodeMirror?

**Comment by jcubic** [comment]
I'm not sure but I don't see only

```javascript
state.localState = CodeMirror.startState(jsMode);
```

I don't see anywhere ending the mode in PEG.js (no "}" check anywhere). Maybe that is the problem. But I'm not sure how to turn off the inner mode, there is no `endState` in CodeMirror.

**Comment by marijnh** [comment]
There is no file named PEG.js in the CodeMirror distribution. Is this about pegjs.js, or some 3rd party mode?

**Comment by jcubic** [comment]
About default pegjs.js mode that is included in the library. The name for the parser generator is PEG.js so I used the name of the project not the filename.

**Comment by marijnh** [comment]
Your PR looks rather unfinished and confused, and this issue doesn't provide a bug description that I can reproduce, so at this point I can't really do much. Also, that mode was contributed and I'm not willing to invest a lot of time into it.

**Comment by jcubic** [comment]
Yes, the PR is broken, I thought that I've fixed the pegjs mode but I'm not able to make it work the same as htmlmixed. I'm getting stack overflow when trying to get pegjs mode inside pegjs mode. It's really hard because there is no detailed documentation how to write modes. At least I didn't find any.
